src/App.tsx

```diff
@@ -1,14 +1,62 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { FC, useState, useEffect } from 'react';
+import './App.scss';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoStatusOptions } from './enums/TodoStatusOptions';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { LoadingError } from './components/LoadingError';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [selectedOption, setSelectedOption] = useState<string>(TodoStatusOptions.all);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [hasLoadingError, setHasLoadingError] = useState<boolean>(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  async function fetchTodos() {
+    setIsLoading(true);
+
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+      setIsLoading(false);
+    } catch (error) {
+      setIsLoading(false);
+      setHasLoadingError(true);
+    }
+  }
+
+  const visibleTodos = todos.filter(todo => {
+    const title = todo.title.toLowerCase();
+    const search = searchQuery.toLowerCase().trim();
+
+    const foundTodo = title.includes(search);
+
+    switch (selectedOption) {
+      case TodoStatusOptions.all:
+        return foundTodo;
+      case TodoStatusOptions.active:
+        return !todo.completed && foundTodo;
+      case TodoStatusOptions.completed:
+        return todo.completed && foundTodo;
+      default:
+        return foundTodo;
+    }
+  });
```

Comment:
1. Create a separate helper for this
2. You create `const search = searchQuery.toLowerCase().trim();` this const every iteration.